content/docs/hooks/use-slider.mdx

```diff
@@ -0,0 +1,396 @@
+| `getRootProps`       | `PropGetter`                            | A function that takes slider root props and handles changes for the slider.                                     |
+| `getTrackProps`      | `PropGetter`                            | A function that takes slider track props and handles changes for the slider's track.                            |
+| `getInnerTrackProps` | `PropGetter`                            | A function that takes slider inner track style props and handles changes in the slider's inner track style.     |
+| `getThumbProps`      | `PropGetter`                            | A function that takes slider thumb props and handles changes for the slider's thumb.                            |
+| `getMarkerProps`     | `RequiredPropGetter<{ value: number }>` | A function that takes the value of the marker and handle changes for the slider's marker positioning and style. |
+| `getInputProps`      | `{}`                                    | A function to get the props of the input field.                                                                 |
+
+## Usage
+
+```jsx
+function Example() {
+  const CustomComponent = (props) => {
+    const {
+      state,
+      actions,
+      getInnerTrackProps,
+      getInputProps,
+      getMarkerProps,
+      getRootProps,
+      getThumbProps,
+      getTrackProps,
+    } = useSlider(props)
+
+    const max = props.max || 100
+    const proportion = max / 100
+    const breakpoint = useBreakpoint()
+
+    const [showValue, setShowValue] = React.useState(false)
+    const [isOnTabStepTo, setIsOnTabStepTo] = React.useState(false)
+    const [stepByNumber, setStepByNumber] = React.useState(1)
+    const [stepToNumber, setStepToNumber] = React.useState(50)
+    const [sliderMarkers, setSliderMarkers] = React.useState(new Array(3))
+    const [showMobileArrows, setShowMobileArrows] = React.useState(false)
+
+    React.useEffect(() => {
+      for (let i = 0; i < 3; i++) {
+        setSliderMarkers((prev) => {
+          const prevCopy = [...prev]
+          prevCopy[i] = getMarkerProps({ value: (max * ((i + 1) * 25)) / 100 })
+          return prevCopy
+        })
+      }
+    }, [])
+
+    React.useEffect(() => {
+      setShowValue(state.isFocused || state.isDragging)
+    }, [state.isDragging, state.isFocused])
+
+    React.useEffect(() => {
+      if (isOnTabStepTo) {
+        actions.stepTo(stepToNumber)
+        setIsOnTabStepTo(false)
+      }
+    }, [isOnTabStepTo])
+
+    const { onKeyDown: onThumbKeyDown, ...thumbProps } = getThumbProps()
+
+    const returnBasedOnIsSmallDesktopOrTablet = (
+      isSmallDesktopOrTabletOutcome,
+      otherBreakpoint,
+    ) =>
+      ['lg', 'md'].includes(breakpoint)
+        ? isSmallDesktopOrTabletOutcome
+        : otherBreakpoint
+
+    const isMobileOrTablet = (res) => ['base', 'sm', 'md'].includes(res)
```

Comment:
I think this example is wayyy to complex for a typical user trying to understand how to use the hook.

I'd recommend simplifying this.

Reply:
Yeah, I was afraid that was the case 😅 , I just wanted to use every prop in the state and actions, etc.
But I'll simplify by... for example:

- Saving the 3 markers apart instead of the loop
- Not using Grid nor GridItem, deleting 75% of that 'Control Panel'
- Keeping the arrow instruction and that functionality just to show off that one can change what the keys do
- Show the value, isDragging and isFocused as texts only instead of doing some effects with them
- Keeping the reset button maybe